3.2/root/usr/share/container-scripts/mongodb/init-petset-replset.sh

```diff
@@ -0,0 +1,152 @@
+#
+# Uses the following global variables:
+# - MONGODB_SERVICE_NAME (optional, defaults to 'mongodb')
+function find_endpoints() {
+  local service_name="${MONGODB_SERVICE_NAME:-mongodb}"
+
+  # Extract host names from lines like this: "10 33 0 mongodb-2.mongodb.myproject.svc.cluster.local."
+  dig "${service_name}" SRV +search +short | cut -d' ' -f4 | rev | cut -c2- | rev
+}
+
+# TODO: unify this and `mongo_initiate` from common.sh
+# Initializes the replica set configuration. It is safe to call this function if
+# a replica set is already configured.
+#
+# Arguments:
+# - $1: host address[:port]
+#
+# Uses the following global variables:
+# - MONGODB_REPLICA_NAME
+# - MONGODB_ADMIN_PASSWORD
+# - WAIT_PRIMARY_STATUS
+function initiate() {
+  local host="$1"
+
+  if mongo --eval "quit(db.isMaster().setName == '${MONGODB_REPLICA_NAME}' ? 0 : 1)" --quiet; then
+    info "Replica set '${MONGODB_REPLICA_NAME}' already exists, skipping initialization"
+    return
+  fi
+
+  local config="{_id: '${MONGODB_REPLICA_NAME}', members: [{_id: 0, host: '${host}'}]}"
+
+  info "Initiating MongoDB replica using: ${config}"
+  mongo admin --eval "rs.initiate(${config});${WAIT_PRIMARY_STATUS}" --quiet
+
+  info "Creating MongoDB users ..."
+  mongo_create_admin
+  mongo_create_user "-u admin -p ${MONGODB_ADMIN_PASSWORD}"
+
+  info "Successfully initialized replica set"
+}
+
+# Adds a host to the replica set configuration. It is safe to call this function
+# if the host is already in the configuration.
+#
+# Arguments:
+# - $1: host address[:port]
+#
+# Global variables:
+# - MAX_ATTEMPTS
+# - SLEEP_TIME
+# - MONGODB_REPLICA_NAME
+# - MONGODB_ADMIN_PASSWORD
+# - WAIT_PRIMARY_OR_SECONDARY_STATUS
+function add_member() {
+  local host="$1"
+  info "Adding ${host} to replica set ..."
+
+  local script
+  script="
+    for (var i = 0; i < ${MAX_ATTEMPTS}; i++) {
+      var ret = rs.add('${host}');
+      if (ret.ok) {
+        quit(0);
+      }
+      // ignore error if host is already in the configuration
+      if (ret.code == 103) {
+        quit(0);
+      }
+      sleep(${SLEEP_TIME});
+    }
+    printjson(ret);
+    quit(1);
+  "
+
+  # TODO: replace this with a call to `replset_addr` from common.sh, once it returns host names.
+  local replset_addr
+  replset_addr="${MONGODB_REPLICA_NAME}/$(find_endpoints | paste -s -d,)"
+
+  if ! mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" --host "${replset_addr}" --eval "${script}" --quiet; then
+    info "ERROR: couldn't join replica set!"
+    return 1
+  fi
+
+  info "Successfully joined replica set"
+  info "Waiting for becoming a SECONDARY node ..."
+
+  # REVIEW: why do we need to wait for "SECONDARY" status? What if this member
+  # joins and becomes the PRIMARY?
+  local rs_status_out
+  rs_status_out="$(mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" --host "${replset_addr}" --eval "${WAIT_PRIMARY_OR_SECONDARY_STATUS}" --quiet || :)"
+
+  if ! echo "${rs_status_out}" | fgrep -xqs 'SECONDARY'; then
```

Comment:
We call it WAIT_PRIMARY_OR_SECONDARY_STATUS, but we're still checking for only `SECONDARY` in the output, needs to change.